qsp_transfer.py

```diff
@@ -0,0 +1,108 @@
+import os
+import sys
+import traceback
+import json
+import boto3
+import base64
+import time
+import re
+import logging
+
+from ftplib import FTP
+from datetime import datetime, timedelta
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+def get_ftp_credentials():
+    client = boto3.client('secretsmanager')
+    get_secret_value_response = client.get_secret_value(SecretId=os.environ['SECRET_NAME'])
+
+    if 'SecretString' in get_secret_value_response:
+        secret = get_secret_value_response['SecretString']
+    else:
+        secret = base64.b64decode(get_secret_value_response['SecretBinary'])
+
+    return json.loads(secret)
+
+def create_data_file(date):
+    if not re.match(r"\d{2}-\d{2}-\d{4}", date):
+        raise Exception(f"Date parameter value '{date}' does not match expected format DD-MM-YYYY")
+
+    log_group = os.environ['LOG_GROUP_NAME']
```

Comment:
I'd use `getenv` here. In addition, I'd probably do some validation up front before we execute anything and potentially get into ambiguous states

Reply:
I've added a `check_environment_variables` function to validate environment variables upfront using `os.getenv`. 
I think it's worth maintaining the `os.environ` call here as this will raise a `KeyError` exception if execution ever got to this line (i.e. a variable was missed in the `check_environment_variables` function) as `getenv` could return `None` here leading to some undefined behaviour.